### PR TITLE
TASK: Restore functionality of RenderingMode view-helpers

### LINKED ISSUE #4642: Adopt Fluid-ViewHelpers to new RenderingModes
Due to changes to the rendering modes, we need to adopt the corresponding Fluid ViewHelper.

Missing implementation:
- [x] [InEditModeViewHelper](https://github.com/neos/neos-development-collection/blob/9.0/Neos.Neos/Classes/ViewHelpers/Rendering/InEditModeViewHelper.php#L87-L88)
- [x] [InPreviewMode](https://github.com/neos/neos-development-collection/blob/9.0/Neos.Neos/Classes/ViewHelpers/Rendering/InPreviewModeViewHelper.php#L83-L84 )

VH implementation to double check:
- [x] [InBackendViewHelper](https://github.com/neos/neos-development-collection/blob/9.0/Neos.Neos/Classes/ViewHelpers/Rendering/InBackendViewHelper.php#L60-L61)
- [x] [LiveViewHelper](https://github.com/neos/neos-development-collection/blob/9.0/Neos.Neos/Classes/ViewHelpers/Rendering/LiveViewHelper.php#L61-L63)

### PR DESCRIPTION
**Pull request by mficzel**
The functionality of the view-helpers `neos:rendering.inBackend`, `neos:rendering.inEditMode`, `neos:rendering.inPreviewMode` and `neos:rendering.live` is restored. Since the rendering mode is now a global property the optional `node` argument of those view-helpers has been removed.

**Checklist**

- [x] Code follows the PSR-2 coding style
- [x] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

**Review by mhsdesign — APPROVED** [review]
I have no clue about fluid :)

File: Neos.Fusion/Classes/ViewHelpers/FusionContextTrait.php [hunk 1]
```diff
@@ -33,18 +33,19 @@ trait FusionContextTrait
...
         if ($view instanceof FusionAwareViewInterface) {
             $fusionObject = $view->getFusionObject();
             $currentContext = $fusionObject->getRuntime()->getCurrentContext();
             if (isset($currentContext[$variableName])) {
-                $value = $currentContext[$variableName];
+                return $currentContext[$variableName];
+            }
+            $fusionGlobals = $fusionObject->getRuntime()->fusionGlobals;
+            if ($fusionGlobals->has($variableName)) {
+                return $fusionGlobals->get($variableName);
```
Comment (bwaidelich) [inline Neos.Fusion/Classes/ViewHelpers/FusionContextTrait.php:45]: Just wondering:
Is this order of evaluation (context first, then global context) something that is always the same.. And didn't we even consider adding a dedicated method to the runtime in order to centralize this fallback behavior?

Comment (mficzel) [reply Neos.Fusion/Classes/ViewHelpers/FusionContextTrait.php:45]: Context takes precedence over Globals as it done in eel expressions. see https://github.com/neos/neos-development-collection/blob/9.0/Neos.Fusion/Classes/Core/Runtime.php#L751C1-L751C1 

Here this allows access not only to renderingMode but also request via FusionContextTrait.

Comment (mhsdesign) [reply Neos.Fusion/Classes/ViewHelpers/FusionContextTrait.php:45]: We decided against adding a method to the Runtime were one can access both types of context via one api. Only for eel was this custom implemented.

**Review by ahaeslich — APPROVED** [review]

File: Neos.Fusion/Classes/ViewHelpers/FusionContextTrait.php (hunk 1 above)
Comment (mficzel) [reply Neos.Fusion/Classes/ViewHelpers/FusionContextTrait.php:45]: @bwaidelich we will merge this once you consider this solved

**Pull request merged**